groups_test.go

```diff
@@ -1124,3 +1124,51 @@ func TestEditGroupPushRules(t *testing.T) {
 		t.Errorf("Groups.EditGroupPushRule returned %+v, want %+v", rule, want)
 	}
 }
+
+func TestCreateGroupWithAllowedEmailDomainsList(t *testing.T) {
+	mux, client := setup(t)
+
+	mux.HandleFunc("/api/v4/groups",
+		func(w http.ResponseWriter, r *http.Request) {
+			testMethod(t, r, http.MethodPost)
+			fmt.Fprint(w, `{"id": 1, "name": "g", "path": "g", "allowed_email_domains_list" : "example.com"}`)
+		})
+
+	opt := &CreateGroupOptions{
+		Name:                    Ptr("g"),
+		Path:                    Ptr("g"),
+		AllowedEmailDomainsList: Ptr("example.com"),
+	}
+
+	group, _, err := client.Groups.CreateGroup(opt, nil)
+	if err != nil {
+		t.Errorf("Groups.CreateGroup returned error: %v", err)
+	}
+
+	want := &Group{ID: 1, Name: "g", Path: "g", AllowedEmailDomainsList: "example.com"}
+	if !reflect.DeepEqual(want, group) {
+		t.Errorf("Groups.CreateGroup returned %+v, want %+v", group, want)
+	}
+}
+
+func TestUpdateGroupWithAllowedEmailDomainsList(t *testing.T) {
+	mux, client := setup(t)
+
+	mux.HandleFunc("/api/v4/groups/1",
+		func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I think that we should actually test here if the request contain the `allowed_email_domains_list` field, right?